assets/src/domain/eventEditor/ui/datetimes/datesList/actionsMenu/dateEntityMainMenuItem/CopyDate.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { pick } from 'ramda';
+import { __ } from '@wordpress/i18n';
+
+import { Copy } from '@application/ui/layout/entityActionsMenu/entityMenuItems';
+import { useDatetimeContext } from '@edtrHooks/index';
+import { useDatetimeItem } from '@edtrServices/apollo/queries';
+import { useDatetimeMutator } from '@edtrServices/apollo/mutations';
+
+const CopyDate = (props) => {
+	const { id } = useDatetimeContext();
+	const datetime = useDatetimeItem({ id });
+	const { createEntity } = useDatetimeMutator();
+	const newDatetime = pick(
+		['capacity', 'description', 'endDate', 'isPrimary', 'name', 'order', 'reserved', 'sold', 'startDate'],
+		datetime
+	);
+	const onClick = () => createEntity(newDatetime);
```

Comment:
Here you will need to do this:
```ts
import useRelations from '@appServices/apollo/relations/useRelations';
const { getRelations } = useRelations();
const tickets = getRelations({
	entity: 'datetimes',
	entityId: id,
	relation: 'tickets',
});
```

Then 
```ts
const onClick = () => createEntity({...newDatetime, tickets});
```

Reply:
thank you, sir
done: https://github.com/eventespresso/event-espresso-core/pull/2546/commits/ce349879323a4c6d16149d0b6dfcde21eeaf8732

Reply:
Can we plz move that entire process into the onclick callback so that we don't need to do anything (copy date, retrieve relations, etc) until the copy button actually gets clicked.

Reply:
:eye: of the :tiger: 

yet another good observation :+1:

Reply:
done: https://github.com/eventespresso/event-espresso-core/pull/2546/commits/8ce917bd7722cb2ad765fc6812ca85f138c492ff